mlir/include/mlir/Dialect/Vector/IR/VectorAttributes.td

```diff
@@ -82,4 +87,63 @@ def Vector_PrintPunctuation : EnumAttr<Vector_Dialect, PrintPunctuation, "punctu
   let assemblyFormat = "`<` $value `>`";
 }
 
+def Vector_StridedSliceAttr : Vector_Attr<"StridedSlice", "strided_slice">
+{
+  let summary = "strided vector slice";
+
+  let description = [{
+    An attribute that represents a strided slice of a vector.
+
+    *Syntax:*
+
+    ```
+    offset = integer-literal
+    stride = integer-literal
+    size = integer-literal
+    offset-list = offset (`,` offset)*
+
+    // Without sizes (used for insert_strided_slice)
+    strided-slice-without-sizes = offset-list? (`[` offset `:` stride `]`)+
+
+    // With sizes (used for extract_strided_slice)
+    strided-slice-with-sizes = (`[` offset `:` size `:` stride `]`)+
+    ```
+
+    *Examples:*
+
+    Without sizes:
+
+    `[0:1][4:2]`
+
+    - The first dimension starts at offset 0 and is strided by 1
+    - The second dimension starts at offset 4 and is strided by 2
+
+    `[0, 1, 2][3:1][4:8]`
```

Comment:
I find this quite hard to read: why are some dimensions separated in their own `[` `]` group but not the non-strided ones?

In general the implicit size seems also hard to read. I'm not sure how to make it better though right now.

Looking at some examples that go from `{offsets = [1, 1], sizes = [2, 2], strides = [1, 1]}` to `[1:2:1][1:2:1]`, I appreciate the compact aspect but I'm not looking forward to need to keep the doc open always to be able to read it...

Reply:
I admit it's not the most readable (I was trying to match the TODO first). 

I grouped the non-strided offsets so they'd look like a normal extraction vs the strided dims (though from updating the tests multiple non-strided dims do not appear to occur in practice much -- the only case IIRC is in the invalid tests).

---

Possibly, a simpler format could be something like:

`[0, 1, 2, 3:1, 4:8]`

And with a size:

`[1:1, 1:1] size 2x2`

i.e. a list of offsets, optionally followed by a `: stride`, with the size displayed separately.

So,
```mlir
%1 = vector.extract_strided_slice %0[0:2:1][2:4:1] : vector<4x8x16xf32> to vector<2x4x16xf32>
```
Becomes:
```mlir
%1 = vector.extract_strided_slice %0[0:1, 2:1] size 2x4 : vector<4x8x16xf32> to vector<2x4x16xf32>
```
---
And,
```mlir
%2 = vector.insert_strided_slice %0, %1[0][0:1][2:1] : vector<2x4xf32> into vector<16x4x8xf32>
```

Becomes:
```mlir
%2 = vector.insert_strided_slice %0, %1[0, 0:1, 2:1] : vector<2x4xf32> into vector<16x4x8xf32>
```

Reply:
> In general the implicit size seems also hard to read. I'm not sure how to make it better though right now.

I don't have any good suggestion myself. Personally, I find the existing syntax clearer.

Reply:
I think the existing syntax is clearer too. My first thought about `[1:2:1]` is something like 
"iterate on `range(1, 2)` with `strides=1`", though it does not make sense to insert/extract_strided_slice ops. Spelling out `offsets`, `sizes`, and `strides` looks clearer to me.

On the other hand, the existing style looks more consistent with the tensor version. E.g., `tensor.insert_slice %11 into %arg4[%8, 0] [1, 2] [1, 1]`

Reply:
I realised I could preserve the old syntax pretty much exactly with a custom `assemblyFormat`. Updated the patch now.